### ISSUE TITLE
Support `select` range?

### ISSUE THREAD

**Issue by gipsyh** [nushell/nushell#5344]
### Related problem

Is it possiable to move `range` into `select`?

for now: `ls | range 0..2 | select name`

after moved: `ls | select 0..2 name`

### Describe the solution you'd like

Move `range` into `select`

**Comment by fdncred** [comment]
Coping here from Discussions.

I'm not a fan of removing the `range` command. However, I would like to see this range syntax used in several commands like `select`, `first`, `last`, `take`, `skip`, and maybe others. Some of those commands may already support a range.

**Comment by jmoore34** [comment]
or perhaps `get`

**Comment by andrasio** [comment]
The problem here is that `range` is used to return you based on rows. `select` on the other hand is meant to return you the column names you care about.

**Comment by adamkpickering** [comment]
I don't agree that it would be a good idea to combine the `select` and `range` commands. I think it would be wise to have a clear distinction between selecting columns and selecting rows. But the main reason I'm commenting is that **right now `select` can return a row**. For example, try `ls | select 0`. In my opinion, this should throw an error.

Maybe it would help to think of the general problem of "slicing a table" as having a few different "dimensions":
1. Are we selecting by row, or by column?
2. Are we selecting by row/column number, or by row/column name (though AFAIK row names aren't a thing in nushell tables)?
3. Are we selecting one row/column, or multiple rows/columns?
4. If we are selecting one row/column, are we returning a table, or are we returning a row (by row I mean the nushell data type)?

This is very much relevant to the [Looking Ahead section](https://www.nushell.sh/blog/2022-05-03-nushell_0_62.html#looking-ahead) in the latest blog post...

**Comment by fdncred** [comment]
selecting by row used to throw an error but we changed it recently to allow you to select a column or a row, just for convenience.

**Comment by adamkpickering** [comment]
> we changed it recently to allow you to select a column or a row

Obviously it's not my call, but I think that's a mistake. To me, `select` has a strong meaning already that comes from SQL: `SELECT name1, name2 FROM mytable` clearly means "get rows `name1` and `name2` from table `mytable`". Many users are going to come to nushell with this idea in their heads. So when they see `select 0` or `select 1` in a script, they'll think that we're trying to select column 0 or column 1. Doesn't that seem confusing? It certainly does to me.

> just for convenience

I'm afraid I don't see how this is more convenient. The only reason I could see it being more convenient is that someone might have gotten used to typing `select` whenever they want to get some subset of a table. But I'm not sure why that would happen.

It might be helpful to look at existing technologies that do things like this. Pandas and SQL come to mind. I think that most cases you'll see either 1) a clear distinction between commands that slice by row and commands that slice by column, or 2) commands that slice by row and column at the same time, but the rows and columns are selected in clearly distinct arguments. I keep thinking of pandas' `at`, `iat`, `loc` and `iloc` dataframe methods.

Just my $0.02. Either way I think nushell is really neat, and I'm excited to see what you guys do with it :grinning:

**Comment by webbedspace** [comment]
Having just read this issue, I sort of agree with the previous post, but also, I notice that this is also how `get` works, and I feel like A) having `select` synched with `get` is important (insofar as one is an "unwrapping" variant of the other), and B) `get` doesn't imply either row or column by its name. So, I have to say I think the status quo is OK, given both of those considerations.

**Comment by T3sT3ro** [comment]
Hi, also remember about `drop` which would be a reverse of picking rows. Btw it would be nice to have a simple chart which shows a 2D table and operations allowed on each axis, this could be a easy way to remember what command does what.

This is very closely related imo to the #13779

**Comment by cptpiepmatz** [comment]
Some time ago we replaced `range` with `slice`. Maybe merging that into `select` is reasonable.
